entities/src/main/java/org/odk/collect/entities/javarosa/intance/LocalEntitiesFileInstanceParser.kt

```diff
@@ -10,26 +10,37 @@ internal class LocalEntitiesFileInstanceParser(private val entitiesRepositoryPro
     ExternalInstanceParser.FileInstanceParser {
 
     override fun parse(instanceId: String, path: String): TreeElement {
+        return parse(instanceId, path, false)
+    }
+
+    override fun parse(instanceId: String, path: String, partial: Boolean): TreeElement {
         val root = TreeElement("root", 0)
 
         val entitiesRepository = entitiesRepositoryProvider()
         entitiesRepository.getEntities(instanceId).forEachIndexed { index, entity ->
             val name = TreeElement(EntityItemElement.ID)
-            name.value = StringData(entity.id)
-
             val label = TreeElement(EntityItemElement.LABEL)
-            label.value = StringData(entity.label)
-
             val version = TreeElement(EntityItemElement.VERSION)
-            version.value = StringData(entity.version.toString())
 
-            val item = TreeElement("item", index)
+            if (!partial) {
+                name.value = StringData(entity.id)
+                label.value = StringData(entity.label)
+                version.value = StringData(entity.version.toString())
+            }
+
+            val item = TreeElement("item", index, partial)
             item.addChild(name)
```

Comment:
For the moment, we unfortunately need to load the whole "structure" of each `item` into memory as otherwise JavaRosa will fail to initialize due to checks it performs on the existence of the `value` and `label` refs for an `itemset`. In most cases, we could get away with just creating the `name` and `label` elements here. However, it's possible to author a form that uses different `value` and `label` refs the full structure is still needed to maintain full compatibility.

@lognaturel is going to look into whether we can remove (or just move) those checks so that we could get away with even less here.